scripts/main/upload.js

```diff
@@ -942,7 +942,7 @@ upload.start = {
 			// After splitting, the escaped spaces must be replaced by
 			// proper spaces as escaping of spaces is a GUI-only thing to
 			// allow input of several paths into a single input field.
-			data.paths = data.paths.match(/(?:\\ |\S)+/g).map((path) => path.replaceAll("\\ ", " "));
+			data.paths = data.paths.match(/(?:\\ |\S)+/g).map((path) => path.replace(/\\ /g, " "));
```

Comment:
I checked it out and tested it, as I wasn't sure if the number of escape character was right. It is. I tried uploading several directories with spaces. I works.